sdk/core/System.ClientModel/tests/Convenience/ClientResultTests.cs

```diff
@@ -38,46 +38,38 @@ public void GetRawResponseReturnsResponse()
 
     #endregion
 
-    #region OptionalClientResult<T>
-
-    [Test]
-    public void CannotCreateOptionalClientResultFromNullResponse()
-    {
-        Assert.Throws<ArgumentNullException>(() => new MockOptionalClientResult<object>(null, null!));
-        Assert.Throws<ArgumentNullException>(() =>
-        {
-            OptionalClientResult<object> result = ClientResult.FromOptionalValue<object>(null, null!);
-        });
-    }
+    #region Optional results: ClientResult<T?>
 
     [Test]
     public void CanCreateOptionalClientResultFromBool()
     {
         // This tests simulates creation of the result returned from a HEAD request.
 
         PipelineResponse response = new MockPipelineResponse(200);
-        OptionalClientResult<bool> result = ClientResult.FromOptionalValue(true, response);
+        ClientResult<bool?> result = ClientResult.FromOptionalValue<bool?>(true, response);
 
         Assert.IsTrue(result.Value);
-        Assert.IsTrue(result.HasValue);
```

Comment:
Can you call Assert.IsTrue(result.Value.HasValue)?

Reply:
Good catch!  Yes, will fix

Reply:
Ah, I see, not a bug, but an additional test case.  Yes, will add